src/sentry/issues/priority.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import logging
+from enum import Enum, IntEnum
+
+from sentry import features
+from sentry.models.activity import Activity
+from sentry.models.group import Group
+from sentry.models.grouphistory import GroupHistoryStatus, record_group_history
+from sentry.types.activity import ActivityType
+
+
+class PriorityLevel(IntEnum):
+    LOW = 25
+    MEDIUM = 50
+    HIGH = 75
+
+
+PRIORITY_LEVEL_TO_STR: dict[int, str] = {
+    PriorityLevel.LOW: "low",
+    PriorityLevel.MEDIUM: "medium",
+    PriorityLevel.HIGH: "high",
+}
+
+
+class PriorityChangeReason(Enum):
+    ESCALATING = "escalating"
+    ONGOING = "ongoing"
+
+
+PRIORITY_TO_GROUP_HISTORY_STATUS = {
+    PriorityLevel.HIGH: GroupHistoryStatus.PRIORITY_HIGH,
+    PriorityLevel.MEDIUM: GroupHistoryStatus.PRIORITY_MEDIUM,
+    PriorityLevel.LOW: GroupHistoryStatus.PRIORITY_LOW,
+}
+
+logger = logging.getLogger(__name__)
+
+
+def update_priority(
+    group: Group,
+    priority: PriorityLevel,
+    reason: PriorityChangeReason | None = None,
+) -> None:
+    """
+    Update the priority of a group and record the change in the activity and group history.
+    """
+    if group.priority_locked_at is not None:
+        return
+
+    group.update(priority=priority)
+    Activity.objects.create_group_activity(
+        group=group,
+        type=ActivityType.SET_PRIORITY,
+        data={
+            "priority": PRIORITY_LEVEL_TO_STR[priority],
+            "reason": reason,
+        },
+    )
+    record_group_history(group, PRIORITY_TO_GROUP_HISTORY_STATUS[priority])
+
+
+def get_priority_for_escalating_group(group: Group) -> PriorityLevel | None:
+    """
+    Get the priority for a group that is escalating.
+    """
+
+    if not group.priority or group.priority == PriorityLevel.HIGH:
+        return PriorityLevel.HIGH
+    elif group.priority == PriorityLevel.LOW:
+        return PriorityLevel.MEDIUM
+    elif group.priority == PriorityLevel.MEDIUM:
+        return PriorityLevel.HIGH
```

Comment:
we _could_ add a `next` method to the PriorityLevel enum with something like 

```
def next(self):
    v = self.value + 25
    if v > 75: # the value for HIGH
        return PriorityLevel(75)
    return PriorityLevel(v)
```

not sure if that's cleaner, but it _does_ keep that logic out of the escalation method. Open to either approach

Reply:
I'm okay to keep this in the escalation method, but I agree both would work fine. For now it's only something we do on escalation so it's putting it with escalation makes sense to me.